### PR TITLE
fix(workflow): backlog item Priority is silently never set (#1501)

### PR DESCRIPTION
**Pull request by lhpaul**
## Summary

Fixes #1501: `add-backlog-item.sh` / `workflow-lib.sh` set backlog item
Priority to the value `Normal`, which does not exist on this repository's
GitHub Projects board (real options: `Urgent`, `High`, `Medium`, `Low`), so
the Priority field silently ended up unset on every item created without an
explicit, already-valid `--priority` value.

Scope was narrowed by live reproduction evidence gathered immediately before
implementation and posted to [issue #1501 comment](https://github.com/lhpaul/ai-dev-framework-template/issues/1501#issuecomment-5336119449):
`High` already resolves correctly on a clean run, so the issue's "defect 3"
(a suspected third, unexplained `High` failure) does not reproduce and is out
of scope for this PR. The real defects are the inverted `Medium`→`Normal`
alias and the `Normal` default, and the helper is fail-open rather than
silent (it does print a `Warning:` line — the defect is that it doesn't stop
anything).

## Changes

- `scripts/development-workflow/workflow-lib.sh`: removed the inverted
  `Medium` → `Normal` alias in `update_tracker_priority_best_effort`.
  Priority resolution already reads the project's actual field options
  dynamically (no hardcoded value list). Added an opt-in `required` mode to
  `update_tracker_named_field_best_effort`: when the tracker provider and
  project are configured but the requested value cannot be resolved/applied,
  the failure is now a hard error (non-zero return, `Error:`-prefixed
  message) instead of always fail-open. Only `update_tracker_priority_best_effort`
  opts into `required`; `update_tracker_type_best_effort` and
  `update_tracker_size_best_effort` remain unchanged (best-effort), and cases
  where the tracker/project genuinely doesn't apply always stay best-effort
  regardless of `required`.
- `scripts/development-workflow/add-backlog-item.sh`: default `--priority`
  changes from `Normal` to `Medium`; updated `--priority` help text.
- `docs/workflow/development-workflow/protocols/00-add-backlog-item-protocol.md`:
  Priority inference heuristics and example command updated from `Normal` to
  `Medium` (this doc explicitly instructed agents to pass `--priority Normal`
  — left as-is it would keep pointing every future caller at the exact
  removed value).
- `docs/workflow/development-workflow/integrations/github-projects.md`:
  Priority field option table corrected from `Normal` to `Medium`.
- Regression tests in `test-add-backlog-item.sh` and
  `test-workflow-lib-github-projects.sh`: assert a requested priority is
  actually present on the board item afterward (mutation carries the
  resolved option ID for `Medium`/`Urgent`/`High`), and that an unresolvable
  requested priority is a hard error with no mutation sent.
- `CHANGELOG.md`: `[Unreleased]` → `### Fixed` entry.

## Out of scope (explicitly)

- Issue #1501's "defect 3" (`High` also failed) — does not reproduce; no
  third defect exists in the resolution path.
- #1502 (max_cycles) — sibling item in the same batch, handled separately;
  `pr-review-loop.sh` is untouched.

## Verification

- `bash scripts/development-workflow/tests/test-add-backlog-item.sh` — 35/35 passed.
- `bash scripts/development-workflow/tests/test-workflow-lib-github-projects.sh` — 147/147 passed.
- `shellcheck --severity=warning` on all modified `.sh` files — clean.
- `python3 scripts/lint/workflow-shell-guard-lint.py --base-ref origin/develop` — clean.
- `markdownlint-cli2` on CHANGELOG.md and both modified docs — 0 issues.
- `bash scripts/lint/check-changelog-duplicate-headers.sh CHANGELOG.md` — passed (single `### Fixed` header under `[Unreleased]`, link references intact).
- CHANGELOG entry checked for trailing whitespace/blank lines; all modified `.md` files end with a trailing newline (MD047).

## Pre-Submission Self-Review Pass

- Reviewed `git diff origin/develop...HEAD` in full; no stale markers, debug
  output, or TODOs left behind.
- Caller consistency: `update_tracker_priority_best_effort` is the only
  caller of `update_tracker_named_field_best_effort` passing `"required"`;
  `update_tracker_type_best_effort` / `update_tracker_size_best_effort`
  callers and their existing tests are unchanged and still pass.
- Confirmed coverage of every doc location that documented the now-removed
  `Normal` default/alias as intended behavior (protocol 00, github-projects.md
  field table) via `grep -rn "Normal"` across `docs/workflow/development-workflow/`;
  left the framework's tracker-agnostic abstract priority-tier vocabulary
  (README.md, protocol 90 ordering heuristics, linear.md's Linear-priority
  mapping) untouched since those are not the literal GitHub Projects board
  option string this bug is about, and changing them would be out-of-scope
  renaming beyond the confirmed defect.
- Issue-body coverage: all "Still-valid remainder of the fix list" items from
  the issue and its narrowing comment are addressed (alias/default fix,
  board-options resolution — already dynamic, hard-error on unresolvable
  request, help text, regression test). The issue's original "reproduce and
  fix the `High` failure" item is explicitly dropped per the live
  reproduction evidence.
- Not a UI-facing change; no design assets applicable. Not a filter-schema
  addition; no canary test required. Not a sweep/batch/pattern-completeness
  change; no residual-gate script required.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

Co-Authored-By: Claude Opus 5 <noreply@anthropic.com>

### CODE REVIEW

**Comment by lhpaul** [comment]
INFO: Using review.on_draft.runner override from .ai-dev-workflow.local.yaml: [claude]. Original list: [codex]. Reason: this Step 7a cycle is dispatched from a Claude Code subagent (BATCH_CONTEXT=true), which cannot invoke the Codex CLI/skill (cross-runner handoff not supported per Protocol 91's reachability table). Dispatching the Claude Code code-reviewer role against REVIEW.md instead.

**Comment by lhpaul** [comment]
### Step 7a Internal Review Gate Summary

**PR type**: Implementation (two-pass)
**Effective reviewer set**: claude (via `.ai-dev-workflow.local.yaml` override; original `review.on_draft.runner: [codex]` is unreachable from a Claude Code subagent)
**Skipped reviewers**: codex (unreachable from Claude Code subagent — cross-runner CLI handoff not supported)

**Pass 1 (Spec Compliance)**

- claude: APPROVED (0 findings) — implementation matches issue #1501's body and the narrowing live-reproduction comment (comment 5336119449): removed the inverted `Medium`→`Normal` alias, changed the `add-backlog-item.sh` default from `Normal` to `Medium`, added opt-in hard-error (`required`) behavior scoped narrowly to Priority, updated `--priority` help text and the two docs that documented the removed default/alias as intended behavior, added regression tests, and added a CHANGELOG entry. No stale markers, TODO/FIXME, or sibling/caller inconsistencies found (`update_tracker_type_best_effort` / `update_tracker_size_best_effort` and their tests are unchanged). Not a Full Pipeline/Refactor item — no plan operational-assumption re-verification applies.

**Pass 2 (Code Quality)**

- claude: APPROVED (0 findings) — project/stack shell conventions followed; `required` mode is opt-in and narrowly scoped (only the Priority convenience wrapper opts in); edge cases covered: provider-mismatch and no-project-configured cases always stay best-effort (return 0) regardless of `required`, since there's nothing to resolve against in those cases. Tests cover the changed behavior directly: `priority_normal_is_hard_error` / `unresolvable_priority_exits_nonzero` demonstrate the guard failing (non-zero exit, no mutation sent) when the targeted violation (an unresolvable requested value) is present, and `priority_medium_mutation_uses_medium_option` / `happy_path_default_priority_is_medium` demonstrate it passing once a real board value is used — both directions of the planted-violation pattern, at `scripts/development-workflow/workflow-lib.sh:2708` (the "could not resolve field or option" branch) and `scripts/development-workflow/tests/test-workflow-lib-github-projects.sh` / `tests/test-add-backlog-item.sh`. Doc edits (protocol 00, github-projects.md) are value substitutions in already-existing tables/examples, not new content — no intra-file duplication or wording-consistency issues introduced. `shellcheck --severity=warning` and `workflow-shell-guard-lint.py` both clean.

**Verdict**: APPROVED
All passes approved at commit `f31a1a4c`.

**Comment by lhpaul** [comment]
@codex review (review triggered by workflow runner, commit: f31a1a4ca5c1)

**Comment by lhpaul** [comment]
### Automated Reviewer Loop Summary

**Result:** clean — no blocking findings
**Platforms:** pr-agent (clean), codex-github (clean)
**Findings:** 0 blocking, 0 suggestions

**Ready reviewer phase:** No net-new blocker was found after the draft GitHub gate.
**Ready-phase platforms:** codex-github

**Advisory findings (non-blocking):**
- Possible Issue ([view comment](https://github.com/lhpaul/ai-dev-framework-template/pull/1506#issuecomment-5336228769))

*Posted automatically by `pr-review-loop.sh`.*

<details>
<summary>Reviewer-loop history (8 iterations)</summary>


```json
{
  "schema": "reviewer_loop_history.v1",
  "pr_number": 1506,
  "updated_at": "2026-08-19T02:54:36Z",
  "history_status": "available",
  "history_unavailable_reason": "",
  "entries": [
    {
      "iteration": 1,
      "recorded_at": "2026-08-19T01:14:00Z",
      "head_sha": "f31a1a4ca5c15e87bcaa11ab5be851895da79dbb",
      "result": "needs_fixes",
      "reason": "unresolved_review_threads",
      "platforms": [
        "pr-agent (clean)",
        "codex-github (needs_fixes)"
      ],
      "blocking_count": 1,
      "suggestion_count": 0,
      "unresolved_thread_count": 0,
      "late_threads_found": 0,
      "phase_after_clean": {
        "enabled": true,
        "platforms": [
          "codex-github"
        ],
        "started": true,
        "net_new_blocker": true,
        "blocking_platform": "codex-github"
      }
    },
    {
      "iteration": 2,
      "recorded_at": "2026-08-19T01:27:31Z",
      "head_sha": "d30e1fc8822d6de364dae1d18f5e01d55ecba9ec",
      "result": "needs_fixes",
      "reason": "unresolved_review_threads",
      "platforms": [
        "pr-agent (clean)",
        "codex-github (needs_fixes)"
      ],
      "blocking_count": 2,
      "suggestion_count": 0,
      "unresolved_thread_count": 0,
      "late_threads_found": 0,
      "phase_after_clean": {
        "enabled": true,
        "platforms": [
          "codex-github"
        ],
        "started": true,
        "net_new_blocker": true,
        "blocking_platform": "codex-github"
      }
    },
    {
      "iteration": 3,
      "recorded_at": "2026-08-19T01:43:28Z",
      "head_sha": "8bce69d27927b6bf73f57fa6139f00c2197c1b80",
      "result": "needs_fixes",
      "reason": "unresolved_review_threads",
      "platforms": [
        "pr-agent (clean)",
        "codex-github (needs_fixes)"
      ],
      "blocking_count": 1,
      "suggestion_count": 0,
      "unresolved_thread_count": 0,
      "late_threads_found": 0,
      "phase_after_clean": {
        "enabled": true,
        "platforms": [
          "codex-github"
        ],
        "started": true,
        "net_new_blocker": true,
        "blocking_platform": "codex-github"
      }
    },
    {
      "iteration": 4,
      "recorded_at": "2026-08-19T01:50:26Z",
      "head_sha": "83f44c6c2ebf47a463d7bea3d4ac717c384f296f",
      "result": "needs_fixes",
      "reason": "unresolved_review_threads",
      "platforms": [
        "pr-agent (clean)",
        "codex-github (needs_fixes)"
      ],
      "blocking_count": 1,
      "suggestion_count": 0,
      "unresolved_thread_count": 0,
      "late_threads_found": 0,
      "phase_after_clean": {
        "enabled": true,
        "platforms": [
          "codex-github"
        ],
        "started": true,
        "net_new_blocker": true,
        "blocking_platform": "codex-github"
      }
    },
    {
      "iteration": 5,
      "recorded_at": "2026-08-19T02:03:07Z",
      "head_sha": "5d4fec34cbfb992759004be4d7887eabc568b784",
      "result": "needs_fixes",
      "reason": "unresolved_review_threads",
      "platforms": [
        "pr-agent (clean)",
        "codex-github (needs_fixes)"
      ],
      "blocking_count": 1,
      "suggestion_count": 0,
      "unresolved_thread_count": 0,
      "late_threads_found": 0,
      "phase_after_clean": {
        "enabled": true,
        "platforms": [
          "codex-github"
        ],
        "started": true,
        "net_new_blocker": true,
        "blocking_platform": "codex-github"
      }
    },
    {
      "iteration": 6,
      "recorded_at": "2026-08-19T02:15:05Z",
      "head_sha": "a9f0716293cd30e3f417955abe61ff30ae5daa1b",
      "result": "clean",
      "reason": "",
      "platforms": [
        "pr-agent (clean)",
        "codex-github (clean)"
      ],
      "blocking_count": 0,
      "suggestion_count": 0,
      "unresolved_thread_count": 0,
      "late_threads_found": 0,
      "phase_after_clean": {
        "enabled": true,
        "platforms": [
          "codex-github"
        ],
        "started": true,
        "net_new_blocker": false,
        "blocking_platform": ""
      }
    },
    {
      "iteration": 7,
      "recorded_at": "2026-08-19T02:16:19Z",
      "head_sha": "a9f0716293cd30e3f417955abe61ff30ae5daa1b",
      "result": "clean",
      "reason": "",
      "platforms": [
        "pr-agent (clean)",
        "codex-github (clean)"
      ],
      "blocking_count": 0,
      "suggestion_count": 0,
      "unresolved_thread_count": 0,
      "late_threads_found": 0,
      "phase_after_clean": {
        "enabled": true,
        "platforms": [
          "codex-github"
        ],
        "started": true,
        "net_new_blocker": false,
        "blocking_platform": ""
      }
    },
    {
      "iteration": 8,
      "recorded_at": "2026-08-19T02:54:36Z",
      "head_sha": "d50abf27fb475eef00f967f6e1cbbc550c483c57",
      "result": "clean",
      "reason": "",
      "platforms": [
        "pr-agent (clean)",
        "codex-github (clean)"
      ],
      "blocking_count": 0,
      "suggestion_count": 0,
      "unresolved_thread_count": 0,
      "late_threads_found": 0,
      "phase_after_clean": {
        "enabled": true,
        "platforms": [
          "codex-github"
        ],
        "started": true,
        "net_new_blocker": false,
        "blocking_platform": ""
      }
    }
  ]
}
```
</details>

**Comment by lhpaul** [comment]
### Step 7a Internal Review Gate Summary (cycle 2 — addresses codex-github Step 7 finding)

**PR type**: Implementation (two-pass)
**Effective reviewer set**: claude (via `.ai-dev-workflow.local.yaml` override; codex unreachable from this Claude Code subagent, as in cycle 1)
**Skipped reviewers**: codex (unreachable)

**Pass 1 (Spec Compliance)**

- claude: APPROVED (0 findings) — the fix directly and narrowly addresses the codex-github P1 finding (`scripts/development-workflow/add-backlog-item.sh#L168` in the reviewed commit): `gh issue create` no longer runs before the priority value is known to be resolvable. No scope expansion beyond the finding.

**Pass 2 (Code Quality)**

- claude: APPROVED (0 findings) — `workflow_tracker_priority_resolvable` is read-only (no mutation, no issue-specific GraphQL calls — verified by the new `resolvable_check_avoids_issue_lookup` test asserting zero `projectItems` calls), and is permissive on every inconclusive/environmental path (wrong provider, no project configured, project/field lookup failure), matching the existing carve-outs in `update_tracker_named_field_best_effort`. New unit tests cover: valid value resolves, confirmed-invalid value blocks, wrong provider stays permissive, no-project stays permissive. Updated the existing `unresolvable_priority_*` test to assert `gh issue create` is never invoked for a confirmed-bad value. `shellcheck --severity=warning` and `workflow-shell-guard-lint.py` both clean. CHANGELOG entry updated in place (fix to unreleased work in this PR, not a new entry, per repo convention).

**Verdict**: APPROVED
All passes approved at commit `d30e1fc8`.

**Comment by lhpaul** [comment]
@codex review (review triggered by workflow runner, commit: d30e1fc8822d)

File: scripts/development-workflow/add-backlog-item.sh [hunk 1]
```diff
@@ -151,12 +155,16 @@ create_cmd() {
...
+    # return 0). update_tracker_priority_best_effort is a hard error (non-zero exit) when the
+    # tracker provider/project are configured but the requested priority value cannot be resolved
+    # against the board's actual Priority field options — under `set -euo pipefail` this aborts
+    # the script so a bad or stale priority value is never silently dropped.
     if [ -n "$type_label" ]; then
       update_tracker_type_best_effort "$issue_number" "$type_label"
     fi
-    local effective_priority="${priority:-Normal}"
+    local effective_priority="${priority:-Medium}"
     update_tracker_priority_best_effort "$issue_number" "$effective_priority"
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply scripts/development-workflow/add-backlog-item.sh]: Fixed at commit d30e1fc8: `add-backlog-item.sh` now calls the new `workflow_tracker_priority_resolvable()` (no-mutation, no issue-specific lookups) to validate the effective priority against the project's real Priority field options BEFORE calling `gh issue create`. An unresolvable explicit value is now rejected with no issue created (exit 1) — see the `unresolvable_priority_skips_issue_create` regression test asserting `gh issue create` is never invoked for that case.

File: scripts/development-workflow/add-backlog-item.sh [hunk 2]
```diff
@@ -115,6 +119,23 @@ create_cmd() {
...
+    # update below, after `gh issue create` had already created the issue —
+    # so a caller that retries on non-zero exit without inspecting stdout
+    # could create a duplicate issue for every retry (see issue #1501 code
+    # review). workflow_tracker_priority_resolvable is a no-op read-only
+    # check (no mutation); it returns 0 (don't block) whenever the tracker
+    # provider/project don't apply or the check itself is inconclusive, so
+    # this pre-check cannot introduce new false rejections — only a
+    # confirmed-invalid value blocks issue creation here.
+    local effective_priority="${priority:-Medium}"
+    if ! workflow_tracker_priority_resolvable "$effective_priority"; then
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply scripts/development-workflow/add-backlog-item.sh]: Fixed at commit 8bce69d2: the implicit default is no longer a static "Medium" literal. `workflow_tracker_default_priority_value()` now adapts to the configured board — prefers "Medium", falls back to "Normal" for boards still set up per the framework's pre-#1501 setup docs (github-projects.md previously told every template consumer to create a "Normal" option), and leaves Priority unset only when a successfully-read board confirms neither candidate exists, the same way an omitted --size/--type is left unset. See `default_falls_back_to_normal_when_medium_absent` / `normal_only_board_default_*` regression tests.

File: scripts/development-workflow/add-backlog-item.sh [hunk 3]
```diff
@@ -151,12 +172,20 @@ create_cmd() {
...
+    # (an invalid effective_priority) is already caught by workflow_tracker_priority_resolvable
+    # above, before the issue was created; this call remains a required safety net for the rarer
+    # cases the pre-check cannot see (e.g. the issue unexpectedly missing from the board, or a
+    # transient GraphQL write failure) — those necessarily surface after issue creation, at which
+    # point the issue URL has already been printed to stdout above.
     if [ -n "$type_label" ]; then
       update_tracker_type_best_effort "$issue_number" "$type_label"
     fi
-    local effective_priority="${priority:-Normal}"
     update_tracker_priority_best_effort "$issue_number" "$effective_priority"
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply scripts/development-workflow/add-backlog-item.sh]: Fixed at commit 8bce69d2: the post-creation required Priority update failure (the rarer case the pre-flight check cannot see — item unexpectedly missing from the board, transient GraphQL write failure) now exits with a distinct code (5, documented in `usage()`) and an explicit "issue was already created ... Do NOT retry issue creation" stderr message, instead of a generic `exit 1` indistinguishable from total failure. See the `partial_success_*` regression tests.

**Comment by lhpaul** [comment]
### Step 7a Internal Review Gate Summary (cycle 3 — addresses codex-github Step 7 cycle-2 findings)

**PR type**: Implementation (two-pass)
**Effective reviewer set**: claude (via `.ai-dev-workflow.local.yaml` override; codex unreachable from this Claude Code subagent)
**Skipped reviewers**: codex (unreachable)

**Pass 1 (Spec Compliance)**

- claude: APPROVED (0 findings) — both codex-github P1/P2 findings from cycle 2 are addressed directly and narrowly: (1) the implicit default now adapts to the board (Medium → Normal fallback → unset) instead of assuming a single literal, so downstream repos whose board is still configured per the pre-#1501 docs (which told them to create a "Normal" option) are not broken by this PR; (2) the post-creation required-update failure now exits with a distinct code (5) and an explicit no-retry message instead of a generic failure. No scope expansion.

**Pass 2 (Code Quality)**

- claude: APPROVED (0 findings) — extracted `_workflow_tracker_priority_field_json` to share the board-lookup between `workflow_tracker_priority_resolvable` and the new `workflow_tracker_default_priority_value`, avoiding duplicating the 4-step provider/project/field lookup. New direct unit tests cover: default prefers Medium, falls back to Normal, empty when neither candidate exists (confirmed board read), and permissive on provider mismatch. New `add-backlog-item.sh` integration tests cover a Normal-only board, a neither-Medium-nor-Normal board, and the distinct partial-success exit path (issue created, URL printed, exit 5, explicit no-retry message). `shellcheck --severity=warning` and `workflow-shell-guard-lint.py` both clean. All 3 codex-github review threads replied to with the specific commit that fixes each, and resolved via the GitHub resolve API.

**Verdict**: APPROVED
All passes approved at commit `8bce69d2`.

**Comment by lhpaul** [comment]
@codex review (review triggered by workflow runner, commit: 8bce69d27927)

File: scripts/development-workflow/add-backlog-item.sh [hunk 4]
```diff
@@ -151,13 +198,28 @@ create_cmd() {
...
+    # `create` and mint a duplicate issue (see issue #1501 code review).
     if [ -n "$type_label" ]; then
       update_tracker_type_best_effort "$issue_number" "$type_label"
     fi
-    local effective_priority="${priority:-Normal}"
-    update_tracker_priority_best_effort "$issue_number" "$effective_priority"
+    if [ -n "$effective_priority" ]; then
+      if ! update_tracker_priority_best_effort "$issue_number" "$effective_priority"; then
+        echo "Error: issue #${issue_number} was already created (${issue_url}) — the required post-creation Priority update failed (see the Error above). Do NOT retry issue creation; instead retry only the Priority update for issue #${issue_number}, or set it manually on the project board." >&2
+        exit 5
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply scripts/development-workflow/add-backlog-item.sh]: Fixed at commit 83f44c6c: the exit-5 signal is now deferred via a `priority_update_failed` flag so Type and Size updates still run when requested, regardless of whether Priority failed. Only after every requested field has had its chance to run does the script raise the partial-success exit. See the new `partial_success_still_updates_size` / `partial_success_still_updates_type` regression tests.

**Comment by lhpaul** [comment]
### Step 7a Internal Review Gate Summary (cycle 4 — addresses codex-github cycle-3 finding)

**PR type**: Implementation (two-pass)
**Effective reviewer set**: claude (via `.ai-dev-workflow.local.yaml` override; codex unreachable from this Claude Code subagent)
**Skipped reviewers**: codex (unreachable)

**Pass 1 (Spec Compliance)**: APPROVED (0 findings) — directly addresses the codex-github P2 finding ("Continue requested field updates after Priority failure"): Type/Size updates are no longer skipped by an early `exit 5` on Priority failure.

**Pass 2 (Code Quality)**: APPROVED (0 findings) — minimal, targeted change (defer exit via a flag); new regression test asserts both Size and Type mutations fire when Priority fails. `shellcheck --severity=warning` and `workflow-shell-guard-lint.py` both clean. All 4 codex-github review threads across 3 cycles replied to with the fixing commit and resolved via the resolve API.

**Verdict**: APPROVED
All passes approved at commit `83f44c6c`.

**Comment by lhpaul** [comment]
@codex review (review triggered by workflow runner, commit: 83f44c6c2ebf)

File: docs/workflow/development-workflow/protocols/00-add-backlog-item-protocol.md [hunk 5]
```diff
@@ -121,11 +124,11 @@ Pass inferred values directly to the helper:
 ./scripts/development-workflow/add-backlog-item.sh create \
   --title "..." \
   --body-file - \
-  --priority Normal \
+  --priority Medium \
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply docs/workflow/development-workflow/protocols/00-add-backlog-item-protocol.md]: Fixed at commit 5d4fec34: protocol 00's Priority inference heuristics and example command now tell agents to OMIT --priority for the routine/default case (letting workflow_tracker_default_priority_value resolve it safely against whatever the board supports) instead of hardcoding --priority Medium. Urgent/High/Low remain safe to pass explicitly since those literals are common to both the current and legacy board vocabularies.

**Comment by lhpaul** [comment]
### Step 7a Internal Review Gate Summary (cycle 5 — addresses codex-github cycle-4 finding)

**PR type**: Non-implementation single-file doc correction within an implementation PR (still treated as two-pass since the PR overall is `fix/*`)
**Effective reviewer set**: claude (via `.ai-dev-workflow.local.yaml` override; codex unreachable from this Claude Code subagent)
**Skipped reviewers**: codex (unreachable)

**Pass 1 (Spec Compliance)**: APPROVED (0 findings) — directly addresses the codex-github P1 finding: protocol 00's authoritative example was itself bypassing the adaptive-default fix. Grepped the repo for any other stale `--priority Medium`/`Normal` guidance (`.agents/`, `.codex/`, `.cursor/`, `.claude/`) — none found.

**Pass 2 (Code Quality)**: APPROVED (0 findings) — doc-only change; `markdownlint-cli2` clean; no shell code touched in this round. All 5 codex-github review threads across 4 fix cycles replied to with the fixing commit and resolved via the resolve API.

**Verdict**: APPROVED
All passes approved at commit `5d4fec34`.

**Comment by lhpaul** [comment]
@codex review (review triggered by workflow runner, commit: 5d4fec34cbfb)

File: docs/workflow/development-workflow/integrations/github-projects.md [hunk 6]
```diff
@@ -91,7 +91,7 @@ Add these custom fields to the project (via project settings UI or GraphQL):
 
 | Field    | Type                                                         | Purpose                                                                                 |
 | -------- | ------------------------------------------------------------ | --------------------------------------------------------------------------------------- |
-| Priority | Single select: `Urgent`, `High`, `Normal`, `Low`             | Drives orchestrator prioritization                                                      |
+| Priority | Single select: `Urgent`, `High`, `Medium`, `Low`             | Drives orchestrator prioritization                                                      |
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply docs/workflow/development-workflow/integrations/github-projects.md:94]: Fixed at commit a9f07162: README.md and Protocol 90's abstract Priority-ordering rules and Proposed Start Batch template (lines 374 / 465 / 856 / 925 / 2032-2033) now list Normal/Medium as an equal-rank pair, matching workflow-batch-overlap.sh's PRIORITY_RANK table which already ranked both equally (no code change needed there). Left the archived plan doc under docs/specs/developments/ unchanged since it is a historical record, not a live protocol contract.

**Comment by lhpaul** [comment]
### Step 7a Internal Review Gate Summary (cycle 6 — addresses codex-github cycle-5 finding)

**PR type**: Doc-only follow-up within the implementation PR
**Effective reviewer set**: claude (via `.ai-dev-workflow.local.yaml` override; codex unreachable from this Claude Code subagent)
**Skipped reviewers**: codex (unreachable)

**Pass 1 (Spec Compliance)**: APPROVED — addresses the codex-github P2 finding directly: README.md and Protocol 90's abstract Priority-ordering docs now recognize `Medium` as equal-rank to `Normal`, matching `workflow-batch-overlap.sh`'s existing (unaffected) `PRIORITY_RANK` table. Verified no code change was needed in the script itself since it already ranked both equally.

**Pass 2 (Code Quality)**: APPROVED — doc-only; `markdownlint-cli2` clean on both files; grepped for remaining bare `Urgent...High...Normal...Low` prose and confirmed only an archived, already-implemented plan document remains (intentionally left as a historical record).

**Verdict**: APPROVED
All passes approved at commit `a9f07162`.

**Comment by lhpaul** [comment]
@codex review (review triggered by workflow runner, commit: a9f0716293cd)

**Comment by lhpaul** [comment]
@codex review (review triggered by workflow runner, commit: d50abf27fb47)

**Comment by lhpaul** [comment]
## Orchestrator review of the PR-Agent `Possible Issue` advisory

The reviewer loop classified [this advisory](https://github.com/lhpaul/ai-dev-framework-template/pull/1506#issuecomment-5336228769) as non-blocking. Per the re-assessment on #1493 — advisory-labelled findings on this repo have shipped real defects before — it was verified against the code rather than accepted on its label. **It does not hold.**

### The claim

> the `required` mode is not applied to the `workflow_github_project_id` call earlier in the function — if that fails, the function returns 0 (best-effort) even in required mode

### Why it does not hold

There is no `workflow_github_project_id` call in `update_tracker_named_field_best_effort`. `project_id` is parsed out of the single `item_json` payload returned by `workflow_github_project_item_for_issue`, and **that parse is required-gated**, as is the subsequent emptiness check.

Audited every return path in the function. All 8 substantive failure branches gate on `required`:

| Branch | `required`-gated |
| --- | --- |
| `item_id` parse failure | ✅ `return 1` |
| `project_id` parse failure | ✅ `return 1` |
| `item_id` empty | ✅ `return 1` |
| `project_id` empty | ✅ `return 1` |
| `field_json` lookup failure | ✅ `return 1` |
| `field_id` parse failure | ✅ `return 1` |
| `option_id` parse failure | ✅ `return 1` |
| `field_id`/`option_id` empty | ✅ `return 1` |

Only two branches return 0 unconditionally, and both are *"this helper does not apply"* cases rather than failures:

- provider is not `github_projects`
- no `project_number` configured

Both are called out in the CHANGELOG entry as deliberately best-effort ("cases where the tracker provider or project genuinely does not apply, remain best-effort as before"), and they are consistent with `workflow_tracker_priority_resolvable`, which also fails open when `_workflow_tracker_priority_field_json` cannot produce a reading. That consistency is the point: when the board cannot be read at all, the whole path degrades to a documented no-op; when the board *can* be read, an explicitly requested unresolvable priority hard-fails **before** `gh issue create`.

Worth noting the advisory partially self-corrects mid-paragraph ("This is fine") before asserting the second concern, which is the one that does not match the code.

**Disposition**: acknowledged, no code change. The documented "hard error" contract for `required` mode is honored on every path where a real board reading was obtained.

### Current-head evidence

The advisory was raised before `d50abf27` (a CHANGELOG-only commit removing PR-internal review narration). Re-verified at the current head:

- Reviewer loop: `RESULT=clean` — `pr-agent` clean, `codex-github` clean, including the 30s post-clean recheck (`LATE_THREADS_FOUND=0`).
- All checks at `d50abf27`: `workflow test harnesses`, `Markdown lint`, `PR-Agent review`, `ShellCheck`, `policy` all success; E2E regression skipped by design.
- Unresolved review threads: **0**.
